hashicorp/terraform-provider-tfe
# Add Optional Expiry to Auth Tokens

## Description
Allow an optional `expired_at` to be set on authentication API tokens via the provider.

## Testing plan

- Ensure that organization and team API tokens that omit the optional `expired_at` field still work as expected.
- Ensure that an `expired_at` can be set on organization and team API tokens and that once set, the expiration date displays in the UI (`/app/org-name/authentication-tokens`) with the expiration date that was provided.
- Ensure that an invalid `expired_at` cannot be set on organization and team API tokens and that if attempted to be set, an error is returned.

## External links

- [Related go-tfe PR](https://github.com/hashicorp/go-tfe/pull/672)
- [API documentation](https://github.com/hashicorp/terraform-docs-common/pull/321)

## Output from acceptance tests

#### `resource_tfe_organization_token_test.go` results:

```
=== RUN   TestAccTFEOrganizationToken_basic
--- PASS: TestAccTFEOrganizationToken_basic (8.99s)
PASS

Process finished with the exit code 0
...
=== RUN   TestAccTFEOrganizationToken_existsWithoutExpiry
--- PASS: TestAccTFEOrganizationToken_existsWithoutExpiry (18.02s)
PASS

Process finished with the exit code 0
...

=== RUN   TestAccTFEOrganizationToken_existsWithExpiry
--- PASS: TestAccTFEOrganizationToken_existsWithExpiry (9.48s)
PASS

Process finished with the exit code 0
```

#### `resource_tfe_team_token_test.go` results:

```
=== RUN   TestAccTFETeamToken_basic
--- PASS: TestAccTFETeamToken_basic (10.22s)
PASS

Process finished with the exit code 0
...
=== RUN   TestAccTFETeamToken_existsWithoutExpiry
--- PASS: TestAccTFETeamToken_existsWithoutExpiry (10.95s)
PASS

Process finished with the exit code 0
...
=== RUN   TestAccTFETeamToken_existsWithExpiry
--- PASS: TestAccTFETeamToken_existsWithExpiry (10.51s)
PASS

Process finished with the exit code 0
```
